signal/signal.go

```diff
@@ -227,3 +252,28 @@ func (c *Interceptor) RequestShutdown() {
 func (c *Interceptor) ShutdownChannel() <-chan struct{} {
 	return c.shutdownChannel
 }
+
+// GetExitCode returns the exit code of the interceptor, which represents
+// the exit status when the main interrupt handler exits. It blocks until
+// the exit code is available.
```

Comment:
Yes since we are waiting for a lock to be available here `c.exitCodeMutex.Lock()`.

Reply:
IMO, I think writing that it is safe for concurrent use would be clearer.

Also, I do not think there is need for a behaviour to block in this function as this function is called when all other subsystems have stopped so the exit code should be readily available then.

Reply:
I agree with you, but since we protect `exitCode` with a mutex, it may be a good practice to wait for it. This is because we return the same `exitCode` in `GetExitCode`, although we're sure that the lock is not held ?

Reply:
I am thinking that maybe the lock might not be needed? https://github.com/lightningnetwork/lnd/pull/8659#discussion_r1587545653